lib/gui/components/safe-webview/safe-webview.js

```diff
@@ -42,75 +53,95 @@ class SafeWebview extends react.PureComponent {
+    this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
+    this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
+    this.refs.webview.addEventListener('new-window', this.constructor.newWindow);
+    this.refs.webview.addEventListener('console-message', this.constructor.consoleMessage);
 
-      // Events React is unaware of have to be handled manually
-      this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
-      this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
-      this.refs.webview.addEventListener('new-window', this.constructor.newWindow);
-      this.refs.webview.addEventListener('console-message', this.constructor.consoleMessage);
+    // Use the 'success-banner' session
+    this.refs.webview.partition = 'persist:success-banner';
 
-    }
+    // It's important that this comes after the partition setting, otherwise it will
+    // use another session and we can't change it without destroying the element again
+    this.refs.webview.src = this.makeURL();
   }
 
   /**
-   * @summary Remove the Webview events if there is an element
+   * @summary Refresh the webview if we are navigating away from the success page
+   * @param {Object} nextProps - upcoming properties
    */
-  componentWillUnmount() {
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.refreshNow && !this.props.refreshNow) {
 
-    // There is no element to remove events from if 'shouldLoad' is false.
-    if (this.state.shouldLoad) {
+      // Reload the page if it hasn't changed, otherwise reset the source URL,
+      // because reload interferes with 'src' setting, resetting the 'src' attribute
+      // to what it was was just prior.
+      if (this.refs.webview.src === this.makeURL()) {
+        this.refs.webview.reload();
 
-      // Events React is unaware of have to be handled manually
-      this.refs.webview.removeEventListener('did-fail-load', this.didFailLoad);
-      this.refs.webview.removeEventListener('did-get-response-details', this.didGetResponseDetails);
-      this.refs.webview.removeEventListener('new-window', this.constructor.newWindow);
-      this.refs.webview.removeEventListener('console-message', this.constructor.consoleMessage);
+      } else {
+        this.refs.webview.src = this.makeURL();
+      }
 
+      this.setState({
+        shouldShow: true
+      });
     }
+  }
+
+  /**
+   * @summary Remove the Webview events
+   */
+  componentWillUnmount() {
 
+    // Events that React is unaware of have to be handled manually
+    this.refs.webview.removeEventListener('did-fail-load', this.didFailLoad);
+    this.refs.webview.removeEventListener('did-get-response-details', this.didGetResponseDetails);
+    this.refs.webview.removeEventListener('new-window', this.constructor.newWindow);
+    this.refs.webview.removeEventListener('console-message', this.constructor.consoleMessage);
```

Comment:
Any (easy) way to avoid the "repetition" between the addEventListener and removeEventListener calls?